### ISSUE TITLE
Common Data Model for Test Server Implementations

### ISSUE THREAD

**Issue by texttechne** [odata2ts/odata2ts#160]
All public available OData services are more or less buggy and don't encompass all the OData functionality. Hence, it would be nice to have custom test server implementations. The following technologies come to mind:
* C# / AspNet (we can borrow from [magic-odata](https://github.com/ShaneGH/magic-odata/tree/main/tests/TestServer) here)
* CAP (typical bookshop example already exists in example folder)
* Olingo

Shortly discussed with @ShaneGH the need to provide a common data model which is suitable to test as many OData features as possible. And this is what this ticket is about: Defining this common data model.

@ShaneGH you already provided the first step in your test implementation with the legendary `OneOfEverything` model, so that all data types are testable (`Edm.Binary` and `Edm.Stream` are special cases for me outside of that model). More thoughts and ideas?

**Comment by ShaneGH** [comment]
Yeah, for sure. The magic-odata model started as a fairly simple blog and grew organically based on use cases.
I'll go through it and pull out link specific parts of it to specific test cases

**Comment by ShaneGH** [comment]
@texttechne to start with, here are some scenarios which would be overkill to code into a server, but would be useful to verify that the code gen works correctly with:

 * `<Schema Namespace="%Namespace has % awkward characters%"`
 * `<EntityContainer Name="%Container has % awkward characters%"`
 * `<EntityContainer Name="%Container has % awkward characters%"`
 * `<EntityType BaseType="A different namespace than the container schema namespace"`
 * `<EntitySet EntityType="A different namespace than the container schema namespace"`
 * `<Singleton EntityType="A different namespace than the container schema namespace"`
 * `<NavigationProperty Type="A different namespace than the container schema namespace...`
 * `<NavigationProperty Type="Collection(A different namespace than the container schema namespace...`
 * `<FunctionImport Function="A different namespace than the container schema namespace...`
 * `<Parameter Type="A different namespace than the container schema namespace...`
 * `<Parameter Type="Collection(A different namespace than the container schema namespace...`
 * `<ReturnType Type="A different namespace than the container schema namespace...`
 * `<ReturnType Type="Collection(A different namespace than the container schema namespace...`

**Comment by texttechne** [comment]
See repo: [test-reference-model](https://github.com/odata2ts/test-reference-model)
